File: c/src/print_prime.c
// build for standalone (for WAVM)
// $ emcc src/print_prime.c src/prime.c -o print_prime_standalone.wasm -O3 -flto -sWASM=1 -sEXPORTED_FUNCTIONS=_prime,_print_prime -sSTANDALONE_WASM=1 --no-entry

#include <stdio.h>
#include "prime.h"

void print_prime(int n)
{
    printf("max prime = %d\n", prime(n));
}

File: go/src/print_prime.go
//go:build js && wasm
// +build js,wasm

// build for standalone (tinygo requires binaryen)
// $ tinygo build -opt=2 -wasm-abi=generic -target=wasi -tags=release -o print_prime_standalone.wasm print_prime.go prime.go

package main

import (
	"fmt"
)

// requires "export" directive for compiler
//export print_prime
func print_prime(n int) void {
	fmt.Println(prime(n))
}
